test/integration/grpcroute_test.go

```diff
@@ -78,10 +81,42 @@ func TestGRPCRouteEssentials(t *testing.T) {
 	require.NoError(t, err)
 	cleaner.Add(gwc)
 
+	t.Log("configuring secret")
+	tlsRouteExampleTLSCert, tlsRouteExampleTLSKey := certificate.MustGenerateSelfSignedCertPEMFormat(certificate.WithCommonName(tlsRouteHostname))
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:       k8stypes.UID("7428fb98-180b-4702-a91f-61351a33c6e8"),
+			Name:      tlsSecretName,
+			Namespace: ns.Name,
+		},
+		Data: map[string][]byte{
+			"tls.crt": tlsRouteExampleTLSCert,
+			"tls.key": tlsRouteExampleTLSKey,
+		},
+	}
+
+	t.Log("deploying secret")
+	secret, err = env.Cluster().Client().CoreV1().Secrets(ns.Name).Create(ctx, secret, metav1.CreateOptions{})
+	require.NoError(t, err)
+	cleaner.Add(secret)
+
 	t.Log("deploying a new gateway")
-	gatewayName := uuid.NewString()
-	gateway, err := DeployGateway(ctx, gatewayClient, ns.Name, gatewayClassName, func(gw *gatewayv1beta1.Gateway) {
-		gw.Name = gatewayName
+	testHostname := "cholpon.example"
+	gateway, err := DeployGateway(ctx, gatewayClient, ns.Name, unmanagedGatewayClassName, func(gw *gatewayv1beta1.Gateway) {
+		gw.Spec.Listeners = []gatewayv1beta1.Listener{
+			builder.NewListener("https").
+				HTTPS().
+				WithPort(ktfkong.DefaultProxyTLSServicePort).
+				WithHostname(testHostname).
+				WithTLSConfig(&gatewayv1beta1.GatewayTLSConfig{
+					CertificateRefs: []gatewayv1beta1.SecretObjectReference{
+						{
+							Name: gatewayv1beta1.ObjectName(secret.Name),
+						},
+					},
+				}).Build(),
+		}
```

Comment:
You can use `IntoSlice` instead of putting this into a slice

https://github.com/Kong/kubernetes-ingress-controller/blob/76f665137a6d55a9b781815128f8a3a7d5836d5f/internal/util/builder/listener.go#L28-L31

BTW why do we add this listener 🤔 and removing it in `DeployGateway`?

Reply:
Ditto other comment: this was the only Listener set that needed a TLS Listener and wasn't already creating its own; changing it was simpler than changing the signature and modifying all the other cases that weren't going to use the HTTPS Listener anyway.

It seems a bit silly to effectively have two variants of Build(), one of which only works if you have exactly one Listener, but whatever.

Reply:
> It seems a bit silly to effectively have two variants of Build(), one of which only works if you have exactly one Listener

It's for convenience, one can be used when you need 1 and the other when you need a slice. Prevents unnecessary conversions